### PR TITLE
Use POST for search

### PR DESCRIPTION
**Pull request by rzezeski**
- Use the POST method for search to prevent a large `fq` parameter
  from causing error.
- Don't pass request headers to Solr as this can break the POST
  search, e.g. if `content-length: 0` is passed.

### CODE REVIEW

**Comment by rzezeski** [comment]
This addresses #221.

**Comment by rzezeski** [comment]
A simple query benchmark on my linux workstation shows a slight bump of ~3.5% in throughput (queries per second), as well as slight reductions in latency. I didn't expect that but nice to see.

|  | Throughput (Ops/s) | Median | 99th (ms) |
| --- | --- | --- | --- |
| GET | 2190 | 14.8 ms | 27.3 ms |
| POST | 2267 | 14.3 ms | 26.6 ms |

File: src/yz_solr.erl [hunk 1]
```diff
@@ -223,11 +223,11 @@ dist_search(Core, Headers, Params) ->
     end.
 
 search(Core, Headers, Params) ->
-    Encoded = mochiweb_util:urlencode(Params),
-    URL = ?FMT("~s/~s/select?~s", [base_url(), Core, Encoded]),
-    Body = [],
+    Body = mochiweb_util:urlencode(Params),
+    URL = ?FMT("~s/~s/select", [base_url(), Core]),
+    Headers2 = [{content_type, "application/x-www-form-urlencoded"}|Headers],
```
Comment (coderoshi) [inline src/yz_solr.erl:228]: Interestingly, if Headers already contains a `content_type`, ibrowse will just convert the header value into a CSV, which is not valid HTTP. But solr also doesn't seem to care, so we're probably good.

**Comment by coderoshi** [comment]
Looks good +1

File: src/yz_solr.erl (hunk 1 above)
Comment (rzezeski) [inline src/yz_solr.erl:228]: Hopefully nothing else is passing `content_type` header to search. I wouldn't think anything is given that search has  been GET up to this point. If it does become an issue we can replace with `lists:keystore`.

**Pull request merged**